src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs

```diff
@@ -0,0 +1,277 @@
+
+            return curTop;
+        }
+
+        /// <summary>
+        /// Inserts a page at the specified index of Navigator.
+        /// If the page is already in Navigator, then it is not inserted.
+        /// </summary>
+        /// <param name="index">The index of Navigator where a page will be inserted.</param>
+        /// <param name="page">The page to insert to Navigator.</param>
+        /// <since_tizen> 8 </since_tizen>
+        public void Insert(int index, Page page)
+        {
+            if ((index < 0) || (index > NavigationPages.Count))
+            {
+                throw new ArgumentOutOfRangeException(nameof(index), "index should be in a range between 0 and the number of pages.");
+            }
+
+            if (page == null)
+            {
+                throw new ArgumentNullException(nameof(page), "page should not be null.");
+            }
+
+            //Duplicate page is not pushed.
+            if (NavigationPages.Contains(page)) return;
+
+            NavigationPages.Insert(index, page);
+            Add(page);
+        }
+
+        /// <summary>
+        /// Inserts a page to Navigator before an existing page.
+        /// If the page is already in Navigator, then it is not inserted.
+        /// </summary>
+        /// <param name="before">The existing page, before which a page will be inserted.</param>
+        /// <param name="page">The page to insert to Navigator.</param>
+        /// <since_tizen> 8 </since_tizen>
+        public void InsertBefore(Page before, Page page)
+        {
+            if (before == null)
+            {
+                throw new ArgumentNullException(nameof(before), "before should not be null.");
+            }
+
+            if (page == null)
+            {
+                throw new ArgumentNullException(nameof(page), "page should not be null.");
+            }
+
+            //Find the index of before page.
+            int beforeIndex = NavigationPages.FindIndex(x => x == before);
+
+            //before does not exist in Navigator.
+            if (beforeIndex == -1)
+            {
+                throw new ArgumentException(nameof(before), "before does not exist in Navigator.");
+            }
+
+            Insert(beforeIndex, page);
+        }
+
+        /// <summary>
+        /// Removes a page from Navigator.
+        /// </summary>
+        /// <param name="page">The page to remove from Navigator.</param>
+        /// <since_tizen> 8 </since_tizen>
+        public void Remove(Page page)
+        {
+            if (page == null)
+            {
+                throw new ArgumentNullException(nameof(page), "page should not be null.");
+            }
+
+            NavigationPages.Remove(page);
+            base.Remove(page);
+        }
+
+        /// <summary>
+        /// Removes a page at the specified index of Navigator.
+        /// </summary>
+        /// <param name="index">The index of Navigator where a page will be removed.</param>
```

Comment:
```suggestion
        /// <param name="index">The index of Navigator where a page will be removed.</param>
        /// <exception cref="ArgumentOutOfRangeException>...</exception>
```

Reply:
I am sorry I should have checked it..
I added the exception tags for each exception case.
Thank you for your review!